frontend/Views/ViewDashboard.svelte

```diff
@@ -45,6 +82,26 @@
         });
     };
 
+    const calculateWidgetStatsKey = function (widget) {
+        return sha1((widget.settings.filter ?? []).join(""));
+    };
+
+    const filterViewForWidget = async function (widget) {
+        if (!widget.settings.filter) {
+            return view;
+        }
+        let tests = await resolveViewTests();
+        tests = tests.reduce((acc, test) => {
+            acc[test.id] = test;
+            return acc;
+        }, {});
+        let viewCopy = structuredClone(view);
+        viewCopy.tests = viewCopy.tests.filter((item) => {
+            const test = tests[item.id] ?? {};
+            return ["id", "release_id", "group_id"].some((key) => widget.settings.filter.includes(test[key]));
+        });
+        return viewCopy;
```

Comment:
this way we return view object in case of no filters and view copy in case of filters present.
Maybe we should align it to not cause future issues?
Or even to add new `filteredTests` attribute which would be used in widgets?

Reply:
Yes, I was deliberating whether or not always return a copy, so I think for safety we should always return a copy here. Fixed